### DISCUSSION TITLE
Can I use Malcolm in pure AWS environment?

### DISCUSSION THREAD

**Discussion by sucremad** [cisagov/Malcolm#333] [Q&A] [answered]
Is it possible to use Malcolm in cloud to collect the traffic of my AWS infrastructure?

**Comment by mmguero** [comment, marked as answer]
We've got some cloud deployment capability right now, and this is an area where were are looking to [become more capable](https://github.com/idaholab/malcolm/issues?q=is%3Aissue+is%3Aopen+-label%3Atraining+label%3Acloud) in the future.

We have documentation [here](https://cisagov.github.io/Malcolm/docs/kubernetes.html#Kubernetes) on deploying Malcolm in Kubernetes, with Amazon Elastic Kubernetes deployment covered [here](https://cisagov.github.io/Malcolm/docs/kubernetes-eks.html#KubernetesEKS). There is also documentation for installing Malcolm on a single EC2 instance [here](https://cisagov.github.io/Malcolm/docs/third-party-envs.html#AWSAMI).

These options I've listed so far work for getting Malcolm deployed in the cloud, but don't take full advantage of all the scale-out stuff cloud (and kubernetes in particular) is known for, yet.

However, In addition to the above, a partner organization has been building [this Malcolm Helm chart](https://github.com/idaholab/Malcolm-Helm/) and using it to deploy Malcolm in a much more scalable way in their organization. The core Malcolm team hasn't yet really had a chance to go through it and evaluate it, hence the documentation using the old vanilla-k8s manifests way, but I think that for cloud people who actually know what they're doing (read: not me, at least not yet 🥲 ) that's what they will probably want to use, so I recommend checking that out.

For how you route your traffic to a virtual interface where it can be captured, from my still-limited understanding you do that by using AWS VPC Traffic Mirroring:

* [Mirror traffic: Amazon VPC training workshop](https://www.awsnetworkshops.com/030_module1/mirrortraffic/)
* [AWS Traffic Mirroring](https://medium.com/@yojohndunn/aws-traffic-mirroring-4d1fa60d9d6f)
* [What is traffic mirroring](https://docs.aws.amazon.com/vpc/latest/mirroring/what-is-traffic-mirroring.html)
* [VPC Traffic Mirroring](https://aws.amazon.com/blogs/aws/new-vpc-traffic-mirroring/)

There is some cool [work the Arkime team has been doing](https://github.com/arkime/aws-aio#arkime-cloud-demo) in automating the setup of that mirroring, which may at some point be adapted into Malcolm, but it's not yet. Basically, though, if you can set up VPC traffic mirroring so that whatever the Malcolm container instance is that's doing the capture sees it as a named interface, you can capture on it.

If you go down one of these paths, please let us know here how it's going for you so we can take your lessons learned into account!

**Comment by mmguero** [comment]
@sucremad , does that cover the information you're looking for?

**Reply by sucremad** [reply]
Hi, @mmguero, thank you for your replay.

Since I m very busy recently, I couldn't have a chance to look them. I will update here, if I can make things work.

**Reply by mmguero** [reply]
Sounds good!
